scripts/sct_compute_mtsat.py

```diff
@@ -132,9 +127,49 @@ def get_parser():
     return parser
 
 
-def main():
+def get_tr_and_flipangle(args):
+    """
+    Retrieve metadata from the json sidecar or from the input arguments.
+    :param args:
+    :return: trmt, trpd, trt1, famt, fapd, fat1
+    """
+    # TODO: deal with the code duplication below
+    if args.trmt is None:
+        fname_json = get_json_file_name(args.mt, check_exist=True)
+        trmt = fetch_metadata(fname_json, 'RepetitionTime')
+    else:
+        trmt = args.trmt
+    if args.trpd is None:
+        fname_json = get_json_file_name(args.pd, check_exist=True)
+        trpd = fetch_metadata(fname_json, 'RepetitionTime')
+    else:
+        trpd = args.trpd
+    if args.trt1 is None:
+        fname_json = get_json_file_name(args.t1, check_exist=True)
+        trt1 = fetch_metadata(fname_json, 'RepetitionTime')
+    else:
+        trt1 = args.trt1
+    if args.famt is None:
+        fname_json = get_json_file_name(args.mt, check_exist=True)
+        famt = fetch_metadata(fname_json, 'FlipAngle')
+    else:
+        famt = args.famt
+    if args.fapd is None:
+        fname_json = get_json_file_name(args.pd, check_exist=True)
+        fapd = fetch_metadata(fname_json, 'FlipAngle')
+    else:
+        fapd = args.fapd
+    if args.fat1 is None:
+        fname_json = get_json_file_name(args.t1, check_exist=True)
+        fat1 = fetch_metadata(fname_json, 'FlipAngle')
+    else:
+        fat1 = args.fat1
+    return trmt, trpd, trt1, famt, fapd, fat1
+
+
+def main(argv):
     parser = get_parser()
-    args = parser.parse_args(args=None if sys.argv[1:] else ['--help'])
+    args = parser.parse_args(argv if argv else ['--help'])
```

Comment:
Although, perhaps making `main(args)` consistent across all scripts could be handled in another PR, considering this is inconsistent in many SCT scripts.

Reply:
> I see you're passing argv to main so that test_sct_compute_mtsat.py can call main with arguments.

yes, exactly!

> Did you have a preferred method in mind for this? I vaguely remember seeing one method endorsed in the Wiki, but I can't find it after a quick search.

I defer to the judgement and expertise for the software engineers in the SCT dev team 😉 
there is this [recent wiki page](https://github.com/neuropoly/spinalcordtoolbox/wiki/Programming%3A-Logging#how-logging-appears-to-users-of-sct-scripts), which does call main(arguments), but what happens _inside_  the main is not documented there (it should).

> Although, perhaps making main(args) consistent across all scripts could be handled in another PR, considering this is inconsistent in many SCT scripts.

sounds good!